spec/integration/api_endpoints_spec.rb

```diff
@@ -505,6 +505,32 @@ def await_event_propagation
     it "deletes a queue" do
       q = @channel.queue(queue_name, durable: false)
       subject.delete_queue("/", queue_name)
+      expect { subject.queue_info("/", queue_name) }.to raise_error(Faraday::ResourceNotFound)
+    end
+
+    it "doesn't delete non-empty queue if if-empty is set" do
+      q = @channel.queue(queue_name, durable: false)
+      q.publish("hello")
+      expect do
+        subject.delete_queue("/", queue_name, false, true)
+      end.to raise_error(Faraday::ClientError)
+
+      subject.purge_queue("/", q.name)
+      subject.delete_queue("/", queue_name, false, true)
+      expect { subject.queue_info("/", queue_name) }.to raise_error(Faraday::ResourceNotFound)
+    end
+
+    it "doesn't delete used queue if if-unused is set" do
+      q = @channel.queue(queue_name, durable: false)
+      consumer = q.subscribe
```

Comment:
This line is incredibly confusing, please restore the handler block. I don't want to investigate why it looks this way three years from now, sorry.

Reply:
Sure. I'll add the comment back as well which would explain why this subscription was added.